### DISCUSSION TITLE
Differences between this and IPFS

### DISCUSSION THREAD

**Discussion by woss** [n0-computer/iroh#222] [Q&A] [answered]
Hi, i wonder what the differences are between the functionality of IROH and IPFS and how does IROH planning to implement upcoming features that land in IPFS.

Are you planning to fork away at some point?

thanks.

**Comment by MinmoTech** [comment]
While I'm not affiliated with the project I think I can answer these questions somewhat.


Iroh is an implementation of IPFS, same as Kubo (formerly go-ipfs) and js-ipfs.
Therefore it's very likely that Iroh will implement "upcoming features" if those are part of the specification.

If by "fork away" you mean that Kubo and and Iroh clients will not be able to communicate with each other, that would not be the case.

**Comment by woss** [comment]
Hey, @MinmoTech  thanks for the answer. So if I understand correctly, the iroh will continue implementing the ipfs standard and all the features, it will be a drop-in replacement for the kubo?

also how does it compare to https://rustipfs.com/ ?

**Reply by MinmoTech** [reply]
No, a drop-in replacement is not the same thing as it being a compliant IPFS implementation.
And it also doesn't have to implement *all* features for this.

For example, Iroh could not implement websocket connections and it would still be able to communicate with other IPFS nodes on other transports (this is a bad example because Iroh is using rust-libp2p and supports websockets).

Watch these if you want to learn more:
https://www.youtube.com/watch?v=qPBR2K2X6cs
https://www.youtube.com/watch?v=ZqubiyngH0o

**Reply by woss** [reply]
very interesting videos. @MinmoTech, thanks for sharing :) 

I will consider this as a potential alternative for https://anagolay.network and https://kelp.digital storage layer ( instead of vanilla ipfs . 
Having said that, if any of the creators of Iroh would like to connect/talk it would be awesome 'coz i have plenty of questions and few suggestions 😉

**Reply by b5** [reply]
hey hey! sorrry for the massive delay. We've been hard at work on getting a release out the door. Now that we've cut v0.1.0 I'd be delighted to chat if you have questions @woss!

**Reply by woss** [reply]
Hey, yeah, so much has happened since Sep 3rd :) 

Would be interesting to have a call and see how we can cooperate. That is if you are interested.

**Reply by b5** [reply]
absolutely. mind emailing [my github handle]@n0.computer ?

**Reply by woss** [reply]
email? are we living in 2000?

**Reply by b5** [reply]
yeah ok agreed email isn't going to cut it in the long run. To that end, I've set up an #iroh channel on the IPFS discord server we can use to chat. Announcement & details: #481

**Comment by b5** [comment, marked as answer]
> Hi, i wonder what the differences are between the functionality of IROH and IPFS and how does IROH planning to implement upcoming features that land in IPFS.

We don't implement every feature kubo (formerly go-ipfs) does. a detailed rundown is here: https://iroh.computer/docs/iroh-and-kubo/

> Are you planning to fork away at some point?

No. we plan to work with the IPFS community to advance the protocol along with other implementations. Of course we plan to be the most efficient, but not at the cost of compatibility.

> Hey, @MinmoTech thanks for the answer. So if I understand correctly, the iroh will continue implementing the ipfs standard and all the features, it will be a drop-in replacement for the kubo?

The goal is to be a drop in replacement for all features of kubo we support.
